File: js/runDijkstra.js
import {buildGraph} from "./buildGraph.js"
import {findShortestPath} from "./dijkstra.js"

const Nroot = 300

console.time(`build Graph ${Nroot} * ${Nroot}`)
const g = buildGraph(Nroot)
console.timeEnd(`build Graph ${Nroot} * ${Nroot}`)

console.time("dijkstra")
const s = findShortestPath(g, "0", String(Nroot**2 - 1))
console.timeEnd("dijkstra")

console.log(s);

File: python/runDijkstra.py
from buildGraph import buildGraph
from dijkstra import findShortestPath
import time

Nroot = 100

t0 = time.time()

g = buildGraph(Nroot)

t1 = time.time()
print(f"build Graph {Nroot} x {Nroot}", str(t1 - t0))

shortestPath = findShortestPath(g, 0, Nroot**2 - 1)

t2 = time.time()

print(f"dijkstra", str(t2 - t1))

print(shortestPath)